internal/keystore/azure/error.go

```diff
@@ -22,18 +22,19 @@ type errorResponse struct {
 	} `json:"error"`
 }
 
-// parseErrorResponse parses the response body as
-// KeyVault secrets API error response.
-func parseErrorResponse(resp *http.Response) (errorResponse, error) {
-	const MaxSize = 1 * mem.MiB
-	limit := mem.Size(resp.ContentLength)
-	if limit < 0 || limit > MaxSize {
-		limit = MaxSize
+// transportErrToStatus converts a transport error to a Status.
+func transportErrToStatus(err error) (status, error) {
+	var rerr *azcore.ResponseError
+	if errors.As(err, &rerr) {
+		var errorResponse errorResponse
+		if rerr.RawResponse != nil {
+			json.NewDecoder(rerr.RawResponse.Body).Decode(&errorResponse)
```

Comment:
```suggestion
			json.NewDecoder(rerr.RawResponse.Body).Decode(&errorResponse)
```

Lets not ignore the error from JSON parsing